src/index.ts

```diff
@@ -0,0 +1,34 @@
+import * as core from '@actions/core';
+import { updateCorsConfig } from './lib/update-cors-cloud';
+import { authenticateCloud } from './lib/authenticate-cloud';
+// import { constructUrls } from './lib/construct-urls';
+
+const AM_URL = core.getInput('AM_URL');
+const username = core.getInput('USERNAME');
+const password = core.getInput('PASSWORD');
+const realm = core.getInput('REALM_PATH');
+const origins = core.getInput('origins'); // this is json as input
+const remove = Boolean(core.getInput('remove_origins'));
+
+async function update(): Promise<void> {
+  try {
+    // We first need to authenticate to get an sso token
+    const {
+      data: { tokenId: ssoToken },
+    } = await authenticateCloud({ AM_URL, realm, username, password });
+
+    const originsToAdd = JSON.parse(origins)
+      .map((val: { url: string; private: boolean }) => val?.url ?? '')
+      .filter(Boolean);
```

Comment:
I'm not really sure what this is doing. Could you elaborate on this?

Reply:
Using `Boolean` like this is quite clever, but it took me a good long while to figure it out ;)

Reply:
Can you add that to a code comment explaining the above. I don't want the next poor chap to come along and stare at the code trying to figure out what it does and why.